crates/ruma-common/src/events/room/message/sanitize.rs

```diff
@@ -0,0 +1,209 @@
+//! Convenience methods and types to sanitize text messages.
+
+#[cfg(feature = "unstable-sanitize")]
+mod html_fragment;
+#[cfg(feature = "unstable-sanitize")]
+mod html_sanitizer;
+
+#[cfg(feature = "unstable-sanitize")]
+use html_sanitizer::HtmlSanitizer;
+
+/// Sanitize the given HTML string.
+///
+/// This removes the [tags and attributes] that are not listed in the Matrix specification.
+///
+/// It can also optionally remove the [rich reply fallback].
+///
+/// [tags and attributes]: https://spec.matrix.org/v1.2/client-server-api/#mroommessage-msgtypes
+/// [rich reply fallback]: https://spec.matrix.org/v1.2/client-server-api/#fallbacks-for-rich-replies
+#[cfg(feature = "unstable-sanitize")]
+pub fn sanitize_html(
+    s: &str,
+    mode: HtmlSanitizerMode,
+    remove_reply_fallback: RemoveReplyFallback,
+) -> String {
+    let sanitizer = HtmlSanitizer::new(mode, remove_reply_fallback);
+    sanitizer.clean(s)
+}
+
+/// What HTML [tags and attributes] should be kept by the sanitizer.
+///
+/// [tags and attributes]: https://spec.matrix.org/v1.2/client-server-api/#mroommessage-msgtypes
+#[cfg(feature = "unstable-sanitize")]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[allow(clippy::exhaustive_enums)]
+pub enum HtmlSanitizerMode {
+    /// Keep only the tags and attributes listed in the Matrix specification.
+    Strict,
+
+    /// Keeps all the tags and attributes in `Strict` mode, and others that are not in this section
+    /// of the spec, but might be encountered.
```

Comment:
Feels a bit unclear.

```suggestion
    /// Like `Strict` mode, but also keeps additional tags and attributes for which the spec has not yet included them, but are reasonable to keep.
```

I also see the COMPAT consts/statics are non-public, it may be useful to make them public when moving this to ruma-html, so this snippet can refer to them.